baremetal/src/VMemAllocator.cc

```diff
@@ -118,6 +118,22 @@ ebbrt::VMemAllocator::Alloc(size_t npages, size_t pages_align,
          npages);
 }
 
+// updates page fault handler for Region for specific
+// virtual page
+void ebbrt::VMemAllocator::SetPFH(
+    ebbrt::Pfn vfn, std::unique_ptr<PageFaultHandler> pf_handler) {
+  std::lock_guard<SpinLock> lock(lock_);
+  for (auto it = regions_.begin(); it != regions_.end(); ++it) {
+    if (vfn == it->first) {
+      it->second.set_page_fault_handler(std::move(pf_handler));
+      return;
+    }
+  }
+
+  kabort("%s: unable to find address 0x%lx in regions_ table\n",
+         __PRETTY_FUNCTION__, vfn.ToAddr());
+}
+
```

Comment:
I needed a way to update the page fault handlers for a specific VFN since they were set to be nullptrs. I couldn't do it at the creation in Alloc when the default page fault handler was set up since I needed a  table (below) to know that mapping.